### ISSUE TITLE
Editable install do not use `wheel.install-dir` when `editable.rebuild` is trigerred

### ISSUE THREAD

**Issue by psalvaggio** [scikit-build/scikit-build-core#866]
I am trying to make a dual-purpose package for use with conda that supports consumption both via Python and C++ (via CMake `find_package()`). I also would like to support editable installs. My conda recipe looks like:

1. `pip install -no-deps`
2. Symlink the relevant things (include/ lib64/cmake/$PROJECT_NAME, my libraries) into $CONDA_PREFIX so my project is easily consumable by other things in the venv.

I have 2 avenues to make this work:

1. Use `wheel.install-dir` and make my CMake agnostic to `scikit-build-core`. This is my preferred option. However, I am running into an issue with the editable rebuilds not respecting `install-dir`. When I trigger a rebuild, I see all of my install targets go to `site-packages` instead of `site-packages/<project_name>`.
2. Make my CMake aware of SKBUILD and use that to set a prefix on my install locations. This can be made to work, but I then run into an issue with step 2 of my conda recipe, where symlinks into the CONDA_PREFIX will result in a broken CMake config due to the extra <project_name> in the install paths. I also tried to install to SKBUILD_DATA_DIR directly, bypassing the symlinking step, however, this result in /tmp paths showing up in my installed CMake config files (target-release.cmake generated via install(EXPORT).

Is there a way to make editable installs respect the `wheel.install-dir` option?

**Comment by LecrisUT** [comment]
> 1. Use `wheel.install-dir` and make my CMake agnostic to `scikit-build-core`. This is my preferred option. However, I am running into an issue with the editable rebuilds not respecting `install-dir`. When I trigger a rebuild, I see all of my install targets go to `site-packages` instead of `site-packages/<project_name>`.

Can you elaborate on this. I have a project setting `tool.scikit-build.wheel.install-dir="spglib"` and it correctly populates the CMake installed files inside `site-packages/spglib`. How does your CMake workflow work?

I have also been considering whether to allow the CMake project to install the python bindings outside of the SKBUILD environment, and ultimately I have decided against that because I couldn't find a way to populate the metadata files that would allow to `pip uninstall` cleanly.

**Comment by psalvaggio** [comment]
Sure... my CMake looks roughly like this:
```cmake
install(
    TARGETS ${PROJECT_NAME}
    EXPORT ${PROJECT_NAME}Targets
    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
)

if (SKBUILD EQUAL "2")
  install(TARGETS ${PY_BINDINGS} DESTINATION .)
endif()

# standard CMake config boilerplate
# install(EXPORT) goes to ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
```
when I do this and I `pip install --no-deps --no-build-isolation -e .`, things initially look fine and everything is where I expect. However, when I change something in my source code and trigger a rebuild, things get installed to `site-packages` instead of `site-packages/<project_name>`. My `pyproject.toml` looks like:
```toml
[build-system]
requires = ["scikit-build-core"]
build-backend = "scikit_build_core.build"

[tool.scikit-build]
editable.rebuild = true
build-dir = "build"
wheel.install-dir = "<projecT_name<"
sdist.exclude = [
   "*.cpp",
   "*.c",
   "*.h",
   "CMakeLists.txt",
]
```

**Comment by LecrisUT** [comment]
Great thanks for the elaboration. Indeed I managed to replicate this issue, and indeed this is a bug.

**Comment by LecrisUT** [comment]
@psalvaggio would you want to write a patch for this? The other relevant part is
https://github.com/scikit-build/scikit-build-core/blob/8cbd92f430c20ad5e0d9725198a1397ab599437d/src/scikit_build_core/build/_editable.py#L39-L51
where you need to alter `editable_txt` to include the `wheel.install-dir` in the generated `_<pkg>_editable.py` file

**Comment by henryiii** [comment]
Oops, wrong copy paste.

**Comment by henryiii** [comment]
Should be fixed in 0.10.4!
